File: src/com/oatau/vhdl/psi/VhdlFile.java
package com.oatau.vhdl.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import com.oatau.vhdl.VhdlFileType;
import com.oatau.vhdl.VhdlLanguage;
import org.jetbrains.annotations.NotNull;

/**
 * Created with IntelliJ IDEA.
 * User: SESA185611
 * Date: 8/02/13
 * Time: 3:54 PM
 */
public class VhdlFile extends PsiFileBase {
    public VhdlFile(@NotNull FileViewProvider fileViewProvider) {
        super(fileViewProvider, VhdlLanguage.Language);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return VhdlFileType.INSTANCE;
    }
}

File: src/com/oatau/vhdl/psi/VhdlFile.scala
package com.oatau.vhdl.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.psi.FileViewProvider
import com.oatau.vhdl.{VhdlFileType, VhdlLanguage}
import com.intellij.openapi.fileTypes.FileType
import com.oatau.vhdl.parser.psi.VEntity

/**
 * User: Michael
 * Date: 24/02/13
 * Time: 12:10
 */
class VhdlFile(fvp: FileViewProvider)
  extends PsiFileBase(fvp,VhdlLanguage.Language) {
  def getFileType: FileType = VhdlFileType.INSTANCE

  def entities = findChildrenByClass(classOf[VEntity])
}
